### DISCUSSION TITLE
Issue with Auth Token Management in Codemagic iOS Builds

### DISCUSSION THREAD

**Discussion by sachinbhartiaphro** [codemagic-ci-cd/codemagic-docs#2552] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Expected results

I am using an auth token for all API calls while making iOS builds in Codemagic. However, I have been searching for a symbolic change over the past few days. Can you find an alternative solution and automatically retrieve the auth token? If this is not possible, how can we ensure that the auth token does not change automatically?

Thanks,
Regards
Sachin Bharti
sachin@aphroecs.com

**Comment by dtrdic** [comment, marked as answer]
Hello @sachinbhartiaphro, the Codemagic API access token can be found in the Codemagic UI under Teams > Personal Account > Integrations > Codemagic API > Show. This token does not change and remains constant for each account.
You can store this token as an environment variable if you need to call the REST API from other workflows. This token remains consistent across your builds
